packages/panzer/disc-fe/src/Panzer_IntegrationValues2.hpp

```diff
@@ -149,34 +174,29 @@ namespace panzer {
 
     Array_Point scratch_for_compute_side_measure; // <Point> size: span() == jac.span()
 
-    /**
-     * \brief Swap the ordering of quadrature points in a specified cell.
-     *
-     * \param[in] cell   Cell index
-     * \param[in] a      Quadrature point a
-     * \param[in] b      Quadrature point b
-     */
-    void swapQuadraturePoints(int cell,int a,int b);
-
+    KOKKOS_INLINE_FUNCTION
     static void convertNormalToRotationMatrix(const Scalar normal[3], Scalar transverse[3], Scalar binormal[3]);
-    
-  protected:
 
+    /// This should be a private method, but using lambdas on cuda forces this to be public.
+    void evaluateRemainingValues(const PHX::MDField<Scalar,Cell,NODE,Dim> & in_node_coordinates, const int in_num_cells);
+
+    /// This should be a private method, but using lambdas on cuda forces this to be public.
+    void getCubatureCV(const PHX::MDField<Scalar,Cell,NODE,Dim> & in_node_coordinates, const int in_num_cells);
+
+    /// This should be a private method, but using lambdas on cuda forces this to be public.
+    void generateSurfaceCubatureValues(const PHX::MDField<Scalar,Cell,NODE,Dim> & in_node_coordinates, const int in_num_cells,const SubcellConnectivity & face_connectivity);
```

Comment:
Should we make these anonymous/non-member calls within the cpp file to prevent them from being public?

Reply:
@seamill - that's a neat idea, but I would lean toward keeping them like this. If we make them stand alone, then we have to pass in 10+ more class members (all kokkos views) as arguments to each of the function calls. That seems like a big chance to accidentally mix up views. I'd prefer to leave them like this and hope that cuda fixes their compiler to allow this in the future. Is that ok?

Reply:
Of course, that is fine. I've been running into this issue in other codes, and I'm unsure what the common practice is. As long as they are properly labeled (and they are) I see no issue.